tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -5953,3 +5953,301 @@ def get_prod(arr):
+    test_int16_output = get_norm(integer16)
+
+    assert np.isclose(f_integer16_output, test_int16_output, rtol=rtol, atol=atol)
+    assert matching_types(f_integer16_output, test_int16_output)
+
+    f_integer32_output = f_integer32(integer32)
+    test_int32_output = get_norm(integer32)
+
+    assert np.isclose(f_integer32_output, test_int32_output, rtol=rtol, atol=atol)
+    assert matching_types(f_integer32_output, test_int32_output)
+
+    f_integer64_output = f_integer64(integer64)
+    test_int64_output = get_norm(integer64)
+
+    assert np.isclose(f_integer64_output, test_int64_output, rtol=rtol, atol=atol)
+    assert matching_types(f_integer64_output, test_int64_output)
+
+    f_fl = epyccel(get_norm, language=language)
+    f_fl32 = epyccel(get_norm, language=language)
+    f_fl64 = epyccel(get_norm, language=language)
+
+    f_fl_output = f_fl(fl)
+    test_float_output = get_norm(fl)
+
+    assert np.isclose(f_fl_output, test_float_output, rtol=rtol, atol=atol)
+    assert matching_types(f_fl_output, test_float_output)
+
+    f_fl32_output = f_fl32(fl32)
+    test_float32_output = get_norm(fl32)
+
+    assert np.isclose(f_fl32_output, test_float32_output, rtol=rtol, atol=atol)
+    assert matching_types(f_fl32_output, test_float32_output)
+
+    f_fl64_output = f_fl64(fl64)
+    test_float64_output = get_norm(fl64)
+
+    assert np.isclose(f_fl64_output, test_float64_output, rtol=rtol, atol=atol)
+    assert matching_types(f_fl64_output, test_float64_output)
+
+    f_complex64 = epyccel(get_norm, language=language)
+    f_complex128 = epyccel(get_norm, language=language)
+
+    f_complex64_output = f_complex64(cmplx64)
+    test_complex64_output = get_norm(cmplx64)
+
+    assert np.isclose(f_complex64_output, test_complex64_output, rtol=1e-6, atol=1e-7)
+    assert matching_types(f_complex64_output, test_complex64_output)
+
+    f_complex128_output = f_complex128(cmplx128)
+    test_complex128_output = get_norm(cmplx128)
+
+    assert np.isclose(f_complex128_output, test_complex128_output, rtol=rtol, atol=atol)
+    assert matching_types(f_complex128_output, test_complex128_output)
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [pytest.mark.fortran]),
+        #    pytest.mark.skip(reason="Still under maintenance, See #769")]),
+        pytest.param("c", marks = [
+            pytest.mark.skip(reason="Still under maintenance, See #769"),
+            pytest.mark.c]
+        )
+    )
+)
+
+def test_numpy_norm_array_like_1d(language):
+
+    @types('bool[:]')
+    @types('int[:]')
+    @types('int8[:]')
+    @types('int16[:]')
+    @types('int32[:]')
+    @types('int64[:]')
+    @types('float[:]')
+    @types('float32[:]')
+    @types('float64[:]')
+    @types('complex64[:]')
+    @types('complex128[:]')
+    def get_norm(arr):
+        from numpy.linalg import norm
+        a = norm(arr)
+        return a
+
+    size = 5
+    rtol=1e-6
+    atol=1e-7
```

Comment:
This tolerance is quite high compared to the tolerance in the rest of this file. Do you know why it needs to be so large?

Reply:
Do you know which cases? Maybe there is a problem in the generated code

Reply:
There’s float32 and complex64.

Reply:
Could the difference be related to the precision then? What is the output type for `norm(a)` when a is a `float32[:]`?